google/cloud/internal/opentelemetry_aliases.h

```diff
@@ -34,7 +34,9 @@ using Span = opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>;
 using ScopedSpan = opentelemetry::trace::Scope;
 #else
 using Span = std::nullptr_t;
```

Comment:
Should we give `Span` the same treatment?  Rather than seeing this as just to "appease clang", think of it more as the right thing.  We wouldn't something like `span = scoped_span` to go undetected in some builds.